src/wavesurfer.ts

```diff
@@ -186,12 +186,18 @@ class WaveSurfer extends Player<WaveSurferEvents> {
   }
 
   private initPlayerEvents() {
+    const timeUpdateRerender = () => {
+      const currentTime = this.getCurrentTime()
+      this.renderer.renderProgress(currentTime / this.getDuration(), this.isPlaying())
+      this.emit('timeupdate', currentTime)
+
+      if(this.isPlaying()){
+        requestAnimationFrame(timeUpdateRerender)
+      }
```

Comment:
I already posted a suggestion. You can wrap only the renderProgress call.

Reply:
This sadly doesn't work. I don't think this could cause issues, because I followed what you had in your codebase with the addition to MDN docs.

Reply:
In my code, I intentionally create a (controlled) recursios, so that it triggers an event on each tick whilst playing.
In this case, we don't want recursion because the timeupdate even is already emitted continuously. And not very often btw. So I think the root cause of your problem is somewhere else.

Reply:
The timer is already supposed to be running. It [starts on audio element `play` and stops on `pause`](https://github.com/katspaugh/wavesurfer.js/blob/main/src/wavesurfer.ts#L196-L204).

Reply:
It's supposed to be, but it's not. Try to comment out `this.renderer.renderProgress(currentTime / this.getDuration(), this.isPlaying())` in `initPlayerEvents()` and you'll see that the progress won't be shown until you pause and resume the track.

Reply:
Perfect, thanks for the tip, will update this PR accordingly